### PR TITLE
Adjust new-file-header for dual-license under MPLv2 + GPLv2+

### PR DESCRIPTION
**Pull request by EraYaN**
This gives great future compatibility.

https://www.gnu.org/licenses/license-list.en.html#MPL-2.0

### CODE REVIEW

File: new-file-header.txt [hunk 1]
```diff
@@ -1,22 +1,30 @@
...
+## It provides an explicit per-file license reference that should be present on all new files.
+## To use this header, delete these lines and the following empty line, modify <filename> to
+## the proper full path, and then add new code following the header and a single empty line.
 
 // <filename>
 // Part of the Jellyfin project (https://jellyfin.media)
 //
+//    This Source Code Form is subject to the terms of the Mozilla Public
+//    License, v. 2.0. If a copy of the MPL was not distributed with this file,
+//    You can obtain one at http:#mozilla.org/MPL/2.0/.
```
Comment (JustAMan) [inline new-file-header.txt]: the URL looks strange

File: new-file-header.txt [hunk 2]
```diff
@@ -1,22 +1,30 @@
...
+//    (at your option) any later version.
 //
 //    This program is distributed in the hope that it will be useful,
 //    but WITHOUT ANY WARRANTY; without even the implied warranty of
 //    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 //    GNU General Public License for more details.
 //
 //    You should have received a copy of the GNU General Public License
-//    along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//    along with this program. If not, see <https:#www.gnu.org/licenses/>.
```
Comment (JustAMan) [inline new-file-header.txt]: same

File: LICENSE-MPLv2 [hunk 3]
```diff
@@ -0,0 +1,373 @@
...
+file, then You may include the notice in a location (such as a LICENSE
+file in a relevant directory) where a recipient would be likely to look
+for such a notice.
+
+You may add additional accurate notices of copyright ownership.
+
+Exhibit B - "Incompatible With Secondary Licenses" Notice
+---------------------------------------------------------
+
+  This Source Code Form is "Incompatible With Secondary Licenses", as
```
Comment (JustAMan) [inline LICENSE-MPLv2]: Doesn't this mean "no GPL compatibility"?
Quote from mozilla:
> Parties who release original work under MPL 2.0 may choose to opt out of this compatibility by including a sentence in the license notices that says that the work is “Incompatible With Secondary Licenses.” Any software that includes this notice is not compatible with the GPL or AGPL.

Comment (EraYaN) [reply LICENSE-MPLv2]: But that is why it's not in the header. Cause we want that compatibility.

Comment (JustAMan) [reply LICENSE-MPLv2]: Isn't the license file a "license notice"?

P.S. That quote was not from mozilla but gnu.org, don't want to edit as you've replied already.

Comment (JustAMan) [reply LICENSE-MPLv2]: Hmm, now I read the license file better, so disregard this comment :) Sorry about false alarm.

**Comment by joshuaboniface** [comment]
I've be re-reading the MPL FAQs, and I've come to share similar concerns with @nvllsvm. What are the long-term implications of dual-licensing for one - and does it hurt our user's freedoms? One of, IMO, the GPL's strongest features is its "viral" nature, which MPL doesn't have. For some components, like those required by the iOS app, a dual-license makes sense, but for the whole server and all new code?

**Comment by anthonylavado** [comment]
Another block is looking at what our long term macOS strategy is. I haven’t done the analysis yet, but there may be a concern between the code signing (both Developer ID *and* GateKeeper), when packaging a .app.

Right now, you can easily bypass those requirements, I seem to remember reading that this will be harder in future.

**Comment by EraYaN** [comment]
The MPLv2 does indeed allow for use with commercial software, but under strict guidelines. So for example the original must be included, the file's source must be provided, the copyright notice must be included. For me personally this is not really an issue, the code can never "lose" it's MPL status. So the actual files if they all have the code they can't just take that file and not share modifications, it's just not viral. I would also be okay with the Apache2 license. And that one doesn't even require the source only that the changes are "stated".

And is has no effect on the rest of said software, since the MPLv2 is basically a file based license. So for the other (decompiled code) you will need a separate license. As for users' freedoms, they can choose to license any derivative work under any of the two licenses, arguably giving them more freedom since the GPL can be so restrictive.
Including for example to make a C# based Xamarin app for iOS based on the same models the Server has for it's (new) API.

As for macOS, Apple has not been very forthcoming with "when" unsigned apps are going to be hard to run on macOS. Right now it requires one to first enable non-appstore apps. And then everytime one runs a unsigned/non-identified developer app, it doesn't let you run them but requires you to open the settings app to allow it. (not very nice, but it seems to remember it for *.app packages that do not move and are not updated). It seems like the next version of macOS might do away with that behavior, but nobody really knows, Apple has only sort of mentioned it during WWDC talks.
On the other hand, the project should most likely just sign all our official releases on all platforms that support it anyway, both for convenience and security of the users. That way random "mirrors" can't include nefarious code. And the GPLv2 at least does not require you to release the private key. But a UWP wrapper for the server would be out of the question under de GPL.
And for at least future macOS releases (and don't discount Microsoft either) the same thing would apply as to the iOS release. But it's one of those things never fought out in court so who knows. But lets not be the project that gets taken to court over it.

**Comment by hawken93** [comment]
https://www.gnu.org/licenses/license-list.html#MPL-2.0 can be read for the GPL standpoint in relation to MPL dual licensing

**Pull request closed without merging**